src/mpid/ch4/shm/posix/posix_startall.h

```diff
@@ -23,6 +23,7 @@ static inline int MPIDI_POSIX_mpi_startall(int count, MPIR_Request * requests[])
 
 MPL_STATIC_INLINE_PREFIX void MPIDI_POSIX_prequest_free_hook(MPIR_Request * req)
 {
+    MPIR_Datatype_release_if_not_builtin(MPIDI_POSIX_REQUEST(req)->datatype);
```

Comment:
Now that active message shared memory is merged, this should call `MPIDIG_prequest_free_hook(req);`